lib/blocks.php

```diff
@@ -208,3 +208,55 @@ function do_blocks( $content ) {
 	return $rendered_content;
 }
 add_filter( 'the_content', 'do_blocks', 9 ); // BEFORE do_shortcode().
+
+/**
+ * Determine whether a post has a specific block. This test optimizes for
+ * performance rather than strict accuracy, detecting the pattern of a block type
+ * but not validating its structure. For strict accuracy, you should use the
+ * block parser on post content.
+ *
+ * @see gutenberg_parse_blocks()
+ *
+ * @since 3.4
+ *
+ * @param string $block_name Full Block name to look for.
+ * @param object $post Post.
+ * @return bool Whether the content contain the specified block.
+ */
+function gutenberg_post_has_block( $block_name, $post = null ) {
+
+	if ( null === $post ) {
+		global $post;
+	} else {
+		$post = get_post( $post );
+	}
+
+	return $post && gutenberg_content_has_block( $block_name, $post->post_content );
+}
+
+/**
+ * Determine whether a content string contains a specific block. This test
+ * optimizes for performance rather than strict accuracy, detecting the
+ * block type but not validating its structure. For strict accuracy,
+ * you should use the block parser on post content.
+ *
+ * @since 3.4
+ * @see gutenberg_parse_blocks()
+ *
+ * @param string $block_name Full Block name to look for.
+ * @param string $content Content to test. Fallback on the $post global's content in case it's missing.
+ * @return bool Whether the content contain the specified block.
+ */
+function gutenberg_content_has_block( $block_name, $content = null ) {
```

Comment:
I don't love this magic behavior of `$content = null`. If I passed a `$content` variable that was set to `null`, then I'd unexpectedly get the global `$post->post_content`.

Is there prior art in core that you're basing this on?

Reply:
I was trying to be consistent with `gutenberg_post_has_block` but now I'm having second thoughts.
`gutenberg_post_has_block` is related to `$post` and prior in core, I can think of `has_excerpt`.

But in opposition, I think that `gutenberg_content_has_block` should be specific with params and no magic effects.

I'll make it required. Do you think that `gutenberg_post_has_block` should keep the `global $post` default?

Reply:
> Do you think that `gutenberg_post_has_block` should keep the `global $post` default?

Isn't `gutenberg_post_has_block` a new function? I think it should be passed an explicit string instead of any dependency on the `$post` object.

Taking a step back though, I think we should wait until further progress on #8352 until we spend more time on this. It may be that the parser itself exposes an equivalent method.